srv/modules/runners/validate.py

```diff
@@ -278,37 +278,41 @@ def storage(self):
 
     def ganesha(self):
         """
-        Nodes may only be assigned one ganesha role.  Ganesha depends on 
+        Nodes may only be assigned one ganesha role.  Ganesha depends on
         cephfs or radosgw.
         """
         ganesha_roles = []
         role_mds = False
         role_rgw = False
+        role_ganesha = False
 
-        for node in self.data.keys():
-            if ('roles' in self.data[node]):
-                if('ganesha_configurations' in self.data[node]):
-                    ganesha_roles = list(set(self.data[node].get("roles")) &
-                                        set(self.data[node].get("ganesha_configurations")))
+        for node, data in self.data.items():
+            if ('roles' in data):
+                if('ganesha_configurations' in data):
+                    ganesha_roles = list(set(data.get("roles")) &
+                                        set(data.get("ganesha_configurations")))
                     if len(ganesha_roles) > 1:
                         msg = "minion {} has {} roles. Only one permitted".format(node, ganesha_roles)
                         self.errors.setdefault('ganesha',[]).append(msg)
```

Comment:
User can define custom roles in ganesha_configurations. We are taking intersection of ganesha_configurations and defined roles for node.